src/utils/style-formatter.js

```diff
@@ -1,55 +1,28 @@
 import DEFAULTS from "../style-defaults";
-import colorUtils from "./color-utils";
 import luiIcons from "./lui-icons";
-import { adjustFontSizeBehavior, backgroundPosition, backgroundSize } from "./style-utils";
-import { getImageUrl } from "./url-utils";
+import {
+  adjustFontSizeBehavior,
+  getBackgroundColor,
+  getBackgroundImage,
+  getBorderStyle,
+  getFontStyle,
+  setIconStyle,
+  setLabelSpan,
+} from "./style-utils";
 
 const formatProperty = (path, setting) => `${path}: ${setting};`;
 
 export default {
-  getStyles({ style = {}, disabled, theme, element, app }) {
+  getStyles({ style, disabled, theme, element, app }) {
     let styles =
       "width: 100%;height: 100%;transition: transform .1s ease-in-out;position: absolute;bottom: 0;left: 0; top: 0;right: 0;margin: auto;";
-    const { font = {}, background = {}, border = {} } = style;
```

Comment:
Having or not having these initial values has no effect on button styling values

Reply:
It might actually be needed. For example the test below will fail on your branch but works fine on`master`.

```js
    it("should return default with empty style object", () => {
      const formattedStyle = styleFormatter.getStyles({ style: {}, disabled, theme, element, app });
      expect(formattedStyle).toEqual(defaultStyle);
    });
```

Reply:
... but we have `style `in all possible scenarios, it is never empty or undefined. When the button styles reset via resetAll, we will have some empty properties like:
`style: {
  background: {useImage: false},
  font: {},
  border: {},
  ...
}` 
but the style itself stays with the main properties. This is the reason I think there is no need of them.

Reply:
What about very old buttons. Could it be that these properties were added at a later stage?

Reply:
There will be no issue for those either. For old buttons style properties have also been defined in `object-properties.js` and even by resetting, style will not turn undefined. The same thing happens there, the properties might get empty but by `getStyle` at each section they get a default value in the case they are not defined.

Reply:
What about a mashup where you do something like this:
```js
const nuked = window.stardust.embed(app, {
      types: [
        {
          name: "action-button",
          load: () => Promise.resolve(window["sn-action-button"]),
        },
      ],
    });

    nuked.selections().then((s) => s.mount(document.querySelector(".toolbar")));

    // create a session object
    nuked.render({
      type: "action-button",
      element: document.querySelector(".object"),
      properties: {
        actions: [
          {
            actionType: "selectValues",
            field: "Region",
            value: "Americas;Asia",
            softLock: false,
          },
        ],
        style: {}, // or undefined
      },
    });
```

Reply:
... hadn't thought about mashups 🤔 Umm... then I'll keep them, it seems safer to have them

Reply:
> There will be no issue for those either. For old buttons style properties have also been defined in `object-properties.js` and even by resetting, style will not turn undefined. The same thing happens there, the properties might get empty but by `getStyle` at each section they get a default value in the case they are not defined.

That is not true. The object-properties are used when the button is created the first time. So if those properties did not exist at that time the button will not have them